src/ts/eventdispatcher.ts

```diff
@@ -189,35 +195,68 @@ class EventListenerWrapper<Sender, Args> {
   isOnce(): boolean {
     return this.once;
   }
+
+  clear(): void {
+  }
+}
+
+interface EventAttributes<Sender, Args> {
+  sender: Sender;
+  args: Args;
 }
 
 /**
  * Extends the basic {@link EventListenerWrapper} with rate-limiting functionality.
  */
 class RateLimitedEventListenerWrapper<Sender, Args> extends EventListenerWrapper<Sender, Args> {
 
-  private rateMs: number;
-  private rateLimitingEventListener: EventListener<Sender, Args>;
+  private readonly rateMs: number;
+  private readonly rateLimitingEventListener: EventListener<Sender, Args>;
+
+  // save last seen event attributes
+  private lastSeenEvent: EventAttributes<Sender, Args>;
 
-  private lastFireTime: number;
+  private rateLimitTimout: Timeout;
 
   constructor(listener: EventListener<Sender, Args>, rateMs: number) {
     super(listener); // sets the event listener sink
 
     this.rateMs = rateMs;
-    this.lastFireTime = 0;
 
-    // Wrap the event listener with an event listener that does the rate-limiting
+    // starting limiting the events to the given value
+    const startRateLimiting = () => {
+      this.rateLimitTimout.start();
+    };
+
+    // timout for limiting the events
+    this.rateLimitTimout = new Timeout(this.rateMs, () => {
+      if (this.lastSeenEvent) {
+        this.fireSuper(this.lastSeenEvent.sender, this.lastSeenEvent.args);
+        startRateLimiting(); // start rateLimiting again to keep rate limit active even after firing the last seen event
+        this.lastSeenEvent = null;
+      }
+    });
+
+    // In case the events stopping during the rateLimiting we need to track the last seen one and delegate after the
+    // rate limiting is finished. This prevents missing the last update due to the rate limit.
     this.rateLimitingEventListener = (sender: Sender, args: Args) => {
-      if (Date.now() - this.lastFireTime > this.rateMs) {
-        // Only if enough time since the previous call has passed, call the
-        // actual event listener and record the current time
+      // only fire events if the rateLimiting is not running
+      if (this.shouldFireEvent()) {
         this.fireSuper(sender, args);
-        this.lastFireTime = Date.now();
+        startRateLimiting();
```

Comment:
I think you need to `return;` here, else this fired event will also be set as `lastSeenEvent` and fired again when the timeout times out.